### ISSUE TITLE
Add full IP-XACT list of register/bit access types

### ISSUE THREAD

**Issue by chrishume** [Origen-SDK/o2#29]
Basically expand what's there already (RO, RW, WO, WONCE, RWONCE) to support the full list of access types that IP-XACT supports (20-something, IIRC).

I'll be glad to do the work for this, as I did it on O1.

**Comment by ginty** [comment]
Agreed, and in fact I think our goal should be that they are all modeled behaviorally too.

**Comment by chrishume** [comment]
I _think_ I did what was necessary to model them behaviorally on O1, but I don't know if I spec tested it.  I'll make sure they're modeled correctly on O2 and backfill to O1 if they're not already functional.

**Comment by info-rchitect** [comment]
Why does the register use a different `access` nomenclature than the bit field?

~~~xml
          <spirit:access>read-write</spirit:access>
          <spirit:reset>
            <spirit:value>0x8000</spirit:value>
            <spirit:mask>0xFFFFFFFF</spirit:mask>
          </spirit:reset>
          <spirit:field>
            <spirit:name>second_bit</spirit:name>
            <spirit:description></spirit:description>
            <spirit:bitOffset>14</spirit:bitOffset>
            <spirit:bitWidth>1</spirit:bitWidth>
            <spirit:access>rw</spirit:access>
~~~

**Comment by chrishume** [comment]
I believe it's because the register isaa conglomeration of different bit
field access types.  A register can qualify as ReadWrite because it
contains a mix of ReadOnly and WriteOnly bits.

**Comment by info-rchitect** [comment]
> I believe it's because the register isaa conglomeration of different bit
> field access types.  A register can qualify as ReadWrite because it
> contains a mix of ReadOnly and WriteOnly bits.
> …

Thx Chris.  I guess I just wondered why it didn't use `rw`?
